openssl/src/crypto/rsa.rs

```diff
@@ -18,6 +18,21 @@ impl Drop for RSA {
 }
 
 impl RSA {
+    /// only useful for associating the key material directly with the key, it's safer to use
+    /// the supplied load and save methods for DER formatted keys.
+    pub fn from_public_components(n: BigNum, e: BigNum) -> Result<RSA, SslError> {
+        unsafe {
+            let rsa = try_ssl_null!(ffi::RSA_new());
+            (*rsa).n = n.into_raw();
+            (*rsa).e = e.into_raw();
+            Ok(RSA(rsa))
+        }
+    }
+
+    pub fn from_raw(rsa: *mut ffi::RSA) -> RSA {
```

Comment:
The caller is asserting that the pointer is valid. Otherwise, this would let me segfault in safe code:
```rust
let rsa = RSA::from_raw(ptr::null_mut());
rsa.has_n()
```